### PR TITLE
Deleting the duplicated comment and quick comment fixes [MTT-7677]

### PR DESCRIPTION
**Pull request by Elfi0Kuhndorf**
### Description

I deleted the duplicated comment in ServerPlayerMove to resolve the bug ticket.

Please let me know if I should add a changelog release note line to some minor fix like this one.

### Issue Number(s)

[MTT-7677](https://jira.unity3d.com/browse/MTT-7677)

### Contribution checklist
 - [ ] Tests have been added for the project and/or any internal package
 - [ ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by fernando-cortez** [comment]
Hey Elfi.

> Please let me know if I should add a changelog release note line to some minor fix like this one.

Yeah, it would be nice to document it. Are there any other tasks for removing unnecessary/redundant comments that we could lump into this PR? If not, no worries.

**Review by RikuTheFuffs — DISMISSED** [review]
+1 on doing similar changes (if you're aware of them) in this same PR. I don't think documenting this is necessary: it'd be just noise in the changelog that doesn't add any value to users reading it.

**Comment by Elfi0Kuhndorf** [comment]
I already checked in Client Driven if there are other comments which can be deleted. I am gonna check in the other bitesize samples as well and if yes, then I am gonna add them to the commit.

File: Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs [hunk 1]
```diff
@@ -17,15 +19,22 @@ public void Config(ShipControl owner, int damage, bool bounce, float lifetime)
...
+    {
+        yield return new WaitForSeconds(lifetime);
+        DestroyBullet();
+    }
+
     public override void OnNetworkDespawn()
     {
-        // This is inefficient, the explosion object could be pooled.
-        GameObject ex = Instantiate(explosionParticle, transform.position + new Vector3(0, 0, -2), Quaternion.identity);
+        m_ObjectPool = GameObject.FindWithTag(s_ObjectPoolTag).GetComponent<NetworkObjectPool>();
```
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs:34]: `FindWithTag` is even worse (performance-wise) than a single instantiate, as it searches the entire scene and 

Can you please revert this change?

If you want to use a pool, it should be statically accessible (better) / you should have a direct reference to it (worse)

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs [hunk 2]
```diff
@@ -469,7 +474,7 @@ public void AddBuff(Buff.BuffType buff)
 
         if (buff == Buff.BuffType.Bounce)
         {
-            QuadDamageTimer.Value = NetworkManager.ServerTime.TimeAsFloat + 10;
+            BounceTimer.Value = NetworkManager.ServerTime.TimeAsFloat + 10;
```
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs:477]: Why are we changing the timer we used?

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs [hunk 3]
```diff
@@ -180,11 +180,16 @@ public void TakeDamage(int amount)
 
         if (Health.Value <= 0)
         {
-            Health.Value = 0;
```
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs:183]: Why was this line deleted?

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public class ShipControl : NetworkBehaviour
 
     float m_Acceleration = 12f;
 
-    float m_BulletLifetime = 2;
+    float m_BulletLifetime = 3;
```
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs:44]: Why was this changed?

File: Basic/2DSpaceShooter/Assets/Prefabs/Bullet.prefab
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Prefabs/Bullet.prefab:1]: what are these changes?

File: Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab:1]: what are these changes?

File: Basic/2DSpaceShooter/Assets/Scenes/network.unity
Comment (RikuTheFuffs) [inline Basic/2DSpaceShooter/Assets/Scenes/network.unity:1]: what are these changes?

**Review by RikuTheFuffs — CHANGES_REQUESTED** [review]
It seems that the changes of this PR no longer align with the scope:

It went from a simple documentation PR to a potentially game-breaking thing. If you're touching documentation, there's no need to edit binary files and alter variables that affect gameplay features.

Can you please create a separate PR and JIRA ticket for each non-documentation related change, so that they can be tested properly?

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs (hunk 4 above)
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs:44]: This was changed because the inserting of the coroutine made the bullet disappear after the amount of seconds equal to the lifetime and 2 seconds were too short (it disappeared to early)

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs (hunk 2 above)
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs:477]: Because the bufftype is Bounce so it should be the bounceTimer which should be increased.

File: Basic/2DSpaceShooter/Assets/Prefabs/Bullet.prefab
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Prefabs/Bullet.prefab:1]: I actually did not change anything I will try to revert the change.

File: Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs (hunk 1 above)
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs:34]: I will revert the change. But I would love to knowhow I can make a pool statically accessible?

File: Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab:1]: I added a network object component to the explosion prefab

File: Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs (hunk 1 above)
Comment (RikuTheFuffs) [reply Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs]: You'll need to store a static reference to the NetworkObjectPool somewhere. [Looking at the documentation](https://docs-multiplayer.unity3d.com/netcode/current/advanced-topics/object-pooling/), it seems it's already done by the NetworkObjectPool class itself ( `NetworkObjectPool Singleton` )

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs (hunk 3 above)
Comment (RikuTheFuffs) [reply Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs]: Since Health is a NetworkVariable with events attached to its OnValueChanged, there is something that happens in between: `SetHealthBarValue` gets called.

If `SetHealthBarValue` had some code like:

```
if (Health.Value == 0)
{ 
     Debug.Log("You're dead"); 
     //do other stuff
}
```

The change would break the game.

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs (hunk 4 above)
Comment (RikuTheFuffs) [reply Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs]: Ok, so that relationship should be explained in a comment, and `m_BulletLifetime` should be a static readonly field equal to `2 + k_BulletDisappearTime` (where `k_BulletDisappearTime` is a constant that defines how much it takes for the bullet to disappear.

Without this info in the codebase, only you know about this relationship.

File: Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab
Comment (RikuTheFuffs) [reply Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab:1]: is the explosion just a VFX, or does it have server-side logic? if it is just a VFX, it shouldn't be a NetworkObject.

File: Basic/2DSpaceShooter/Assets/Scenes/network.unity
Comment (RikuTheFuffs) [reply Basic/2DSpaceShooter/Assets/Scenes/network.unity:1]: 1. see comment above
2. Is the NetworkObject pool in a prefab? If so, you should change the prefab, not the scene. This will make sure that the prefab's instances are updated in all scenes, and reduces the chances of merge conflicts with other people doing changes to the scene.

**Review by RikuTheFuffs — APPROVED** [review]

File: Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Prefabs/Explosion.prefab:1]: It has a particle system, but also an audio source on itself

File: Basic/2DSpaceShooter/Assets/Scenes/network.unity
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Scenes/network.unity:1]: It unfortunately is not a prefab, but I could turn it into one?

Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Scenes/network.unity:1]: that would of course still result in a scene change

File: Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs (hunk 3 above)
Comment (Elfi0Kuhndorf) [reply Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs]: I see, you are right it is important to check for that. It does not seem like this is the case here, but I will still add the line back, as it is good visual feedback for the player/user to have the health at 0 before it gets reset.

File: Basic/2DSpaceShooter/Assets/Scenes/network.unity
Comment (RikuTheFuffs) [reply Basic/2DSpaceShooter/Assets/Scenes/network.unity:1]: yeah, the 1st time you do somethign liek this you'll have to touch the scene anyway
